pkg/urbit/include/c/defs.h

```diff
@@ -122,4 +122,30 @@
         }                                                       \
         rut;})
 
+//! Error-handling wrappers for system calls that return >=0 on success
+//! and <0 on failure.
+//!
+//! It's unclear whether `strerror(errno)` is a portable solution for reporting
+//! error messages.
+//!
+//! @param[in] system_call     System call to execute.
+//! @param[in] failure_action  Statement to execute after logging failure.
+#define try_syscall(system_call, failure_action, ...)                          \
+  do {                                                                         \
+    if ( 0 > (system_call) ) {                                                 \
+      fprintf(stderr, "syscall: " __VA_ARGS__);                                \
+      c3_c* msg_c = strerror(errno);                                           \
+      if ( !msg_c ) {                                                          \
+        msg_c = "Unknown error";                                               \
+      };                                                                       \
+      fprintf(stderr,                                                          \
+              " (%s) [%s:%s():%d]\r\n",                                        \
+              msg_c,                                                           \
+              __FILE__,                                                        \
+              __func__,                                                        \
+              __LINE__);                                                       \
+      failure_action;                                                          \
+    }                                                                          \
+  } while ( 0 )
+
```

Comment:
This is a pretty heavyweight macro, I'm not sure how much we want to encourage its use. I'm inclined to put it in `bile.c`, and maybe add some kind of internal/impl header if we need to share it across modules. But this gets to the broader question of a "public api" ...

Reply:
Yeah, I still haven't decided on how I feel about this one. I wrote `try_epoc() `and `try_lmdb()`, which are similar macros for calling `u3_epoc_*` and `mdb_*` functions (respectively), and they've turned out to be super handy, but syscall/libc calls may just be too heterogenenous to paper over with a single macro like this.

Reply:
Moved into `bile.c`.